tck/src/main/java/org/eclipse/microprofile/lra/tck/participant/api/ContextTckResource.java

```diff
@@ -0,0 +1,377 @@
+    @Path(REQUIRED_LRA_PATH)
+    public Response requiredLRA(@HeaderParam(LRA_HTTP_CONTEXT_HEADER) String lraId,
+                                @HeaderParam(LRA_TCK_FAULT_TYPE_HEADER) String tckFaultType,
+                                @HeaderParam(LRA_TCK_FAULT_CODE_HEADER) int tckFaultCode) {
+        setEndPhase(tckFaultType, tckFaultCode);
+
+        return Response.ok().entity(lraId).build();
+    }
+
+    @LRA(value = LRA.Type.REQUIRES_NEW)
+    @PUT
+    @Path(REQUIRES_NEW_LRA_PATH)
+    public Response requiresNew(@HeaderParam(LRA_HTTP_CONTEXT_HEADER) String lraId) {
+        return Response.ok().entity(lraId).build();
+    }
+
+    // test that outgoing calls do not affect the calling context
+    @LRA(value = LRA.Type.REQUIRED)
+    @PUT
+    @Path(CONTEXT_CHECK_LRA_PATH)
+    public Response contextCheck(@HeaderParam(LRA_HTTP_CONTEXT_HEADER) String lraId) {
+        String remote;
+        String active = getActiveLRA();
+        assertEquals("contextCheck1: incoming and active LRAs are different", lraId, active);
+
+        // invoke a remote service which runs in its own context. Do not set the context header
+        remote = restPutInvocation(null, REQUIRES_NEW_LRA_PATH, "");
+        assertNotEquals("contextCheck2: the remote service should not have ran with the active context", remote, active);
+
+
+        assertEquals("contextCheck2: after calling a remote service the active LRAs are different", active, getActiveLRA());
+
+        // invoke a remote service which runs in its own context. Do set the context header
+        remote = restPutInvocation(active, REQUIRES_NEW_LRA_PATH, "");
+        assertNotEquals("contextCheck3: the remote service should not have ran with the active context", remote, active);
+        assertEquals("contextCheck3: after calling a remote service the active LRAs are different", active, getActiveLRA());
+
+        // invoke a remote service which runs in the callers context, ie do set the context header
+        remote = restPutInvocation(active, TCK_NON_PARTICIPANT_RESOURCE_PATH, SUPPORTS_PATH, "");
+        assertEquals("contextCheck4: the remote service should have ran with the active context", remote, active);
+        assertEquals("contextCheck4: after calling a remote service the active LRAs is different", active, getActiveLRA());
+
+        return Response.ok().entity(lraId).build();
+    }
+
+    /**
+     * Query the current value of a metric in the context of an LRA
+     * @param lraId the LRA that owns the metric
+     * @param metric the name of the metric
+     * @return the current value of the metric
+     */
+    @GET
+    @Path(METRIC_PATH + "/{metric}")
+    public Response count(@HeaderParam(LRA_TCK_HTTP_CONTEXT_HEADER) String lraId,
+                          @PathParam("metric") String metric) {
+        if (metrics.containsKey(lraId)) {
+            return Response.ok(metrics.get(lraId).get(metric)).build();
+        }else {
+            return Response.ok(-1).build();
```

Comment:
We don't care since it is a test error to pass in an invalid metric por to pass in the wrong LRA. Both situations will return -1 which will cause the test to fail.